File: web/src/api/intradayConfirmation.ts
import apiClient from './index';
import { toCamelCase } from './utils';
export type Market = 'CN' | 'US';
export type State = 'WAIT' | 'CONFIRMED' | 'FAILED';
export type Source = 'confluence' | 'trend' | 'quant';
export interface Reason { code: string; text: string }
export interface Confirmation {
  code: string; name: string; candidateSource: Source; candidateTradeDate: string;
  candidateReason: string[]; sourceGeneratedAt: string; state: State; confirmationScore: number;
  chaseRisk: 'LOW' | 'MEDIUM' | 'HIGH'; reasons: Reason[]; stateReasons: Reason[];
  metrics: Record<string, number | string | boolean | null>;
  trend: Record<string, number | string | null>;
  firstConfirmedAt: string | null; failedAt: string | null; generatedAt: string | null;
  currentPriceRecovered: boolean;
}
export interface Snapshot {
  market: Market; tradeDate: string; candidateTradeDate: string | null; frozenAt: string | null;
  generatedAt: string | null; status: string; warnings: string[]; items: Confirmation[];
  summary: { total: number; wait: number; confirmed: number; failed: number }; rulesNote: string;
}
const base = '/api/v1/intraday-confirmation';
export const intradayConfirmationApi = {
  async read(market: Market, state?: State, source?: Source): Promise<Snapshot> {
    return toCamelCase((await apiClient.get(base, { params: { market, state, candidate_source: source } })).data);
  },
  async detail(code: string, market: Market): Promise<Confirmation> {
    return toCamelCase((await apiClient.get(`${base}/${encodeURIComponent(code)}`, { params: { market } })).data);
  },
  async run(market: Market): Promise<{ taskId: string }> {
    return toCamelCase((await apiClient.post(`${base}/run`, { market })).data);
  },
};

File: src/finance_analysis/interfaces/api/v1/schemas/intraday_confirmation.py
"""Read-only snapshot contract. A null metric explicitly means unavailable."""

from datetime import date, datetime
from typing import Any, Literal
from pydantic import BaseModel, Field

Market = Literal["CN", "US"]
State = Literal["WAIT", "CONFIRMED", "FAILED"]
Source = Literal["confluence", "trend", "quant"]


class Reason(BaseModel):
    code: str
    text: str


class Confirmation(BaseModel):
    code: str
    name: str
    candidate_source: Source
    candidate_trade_date: date
    candidate_reason: list[str]
    source_generated_at: datetime
    state: State
    confirmation_score: float | None
    available_score_weight: float = 0
    chase_risk: Literal["LOW", "MEDIUM", "HIGH", "UNKNOWN"]
    reasons: list[Reason]
    state_reasons: list[Reason] = Field(default_factory=list)
    metrics: dict[str, Any]
    trend: dict[str, Any]
    score_breakdown: dict[str, float | None] = Field(default_factory=dict)
    first_confirmed_at: datetime | None = None
    failed_at: datetime | None = None
    max_confirmation_score: float = 0
    current_price_recovered: bool = False
    generated_at: datetime | None = None


class Snapshot(BaseModel):
    market: Market
    trade_date: date
    candidate_trade_date: date | None = None
    frozen_at: datetime | None = None
    generated_at: datetime | None = None
    status: str
    warnings: list[str]
    items: list[Confirmation]
    summary: dict[str, int]
    rules_note: str


class RunRequest(BaseModel):
    market: Market = "CN"
